main.go

```diff
@@ -196,7 +196,7 @@ func main() {
 				kernelmech.MECHANISM: chain.NewNetworkServiceClient(kernel.NewClient()),
 			}),
 			sendfd.NewClient(),
-			dnscontext.NewClient(dnscontext.WithChainContext(ctx), dnscontext.WithDNSConfigsMap(dnsConfigsMap)),
+			// dnscontext.NewClient(dnscontext.WithChainContext(ctx), dnscontext.WithDNSConfigsMap(dnsConfigsMap)),
```

Comment:
Check the main branch of this repo. Looks like they are not doing the dns proxying by default. They are relying on a config option. Lets copy the code from the main branch. And set the config to true.

if c.LocalDNSServerEnabled {
		dnsConfigsMap := new(dnsconfig.Map)
		dnsClient = dnscontext.NewClient(dnscontext.WithChainContext(ctx), dnscontext.WithDNSConfigsMap(dnsConfigsMap))

		dnsServerHandler := dnschain.NewDNSHandler(
			checkmsg.NewDNSHandler(),
			dnsconfigs.NewDNSHandler(dnsConfigsMap),
			searches.NewDNSHandler(),
			noloop.NewDNSHandler(),
			cache.NewDNSHandler(),
			fanout.NewDNSHandler(),
		)

		go dnsutils.ListenAndServe(ctx, dnsServerHandler, c.LocalDNSServerAddress)
	}

Reply:
@bharath-avesha and when do we set it to false?

Reply:
Never. The config option would be coming from an env var. For our case, it should always be false. I have edited my prev comment. It should NOT be set to true. It should be false.

Reply:
Please look for the diff that brought in the config flag change in the main branch. Patch the whole diff to our branch.